src/System.Security.Cryptography.Cng/tests/ImportExportTests.cs

```diff
@@ -46,5 +45,191 @@ public static void TestImportExportRoundTrip()
+                {
+                    newCng.ImportParameters(param);
+
+                    // the curve name is not flowed on on explicit export\import (by design) so we are unable to generate new
+                    Assert.ThrowsAny<CryptographicException>(() => newCng.GenerateNewKey());
+                }
+            }
+        }
+
+        [Fact]
+        public static void TestRegenKeyPositive()
+        {
+            if (!EcDSACngTests.IsEccNamedCurveSupported())
+                return;
+
+            ECParameters param, param2;
+            ECDsaCng cng;
+
+            using (cng = new ECDsaCng(ECNamedCurve.brainpoolP160r1))
+            {
+                param = cng.ExportExplicitCurveParameters(true);
+                Assert.True(param.Curve is ECPrimeCurve);
+                Assert.NotEqual(param.D, null);
+
+                cng.GenerateNewKey();
+                param2 = cng.ExportExplicitCurveParameters(true);
+
+                ComparePublicKey(param.Q, param2.Q, false); // public key should not match
+                Assert.False(IsEqual(param.D, param2.D)); // private key should not match
+                CompareCurve(param.Curve, param2.Curve); // curve should match
+            }
+        }
+
+        private static void IsEqual(ECParameters p1, ECParameters p2)
+        {
+            ComparePublicKey(p1.Q, p2.Q);
+            CompareCurve(p1.Curve, p2.Curve);
+        }
+
+        private static void ComparePublicKey(ECPoint q1, ECPoint q2, bool isEqual = true)
+        {
+            Assert.Equal(isEqual, IsEqual(q1.X, q2.X));
+            Assert.Equal(isEqual, IsEqual(q1.Y, q2.Y));
+        }
+
+        private static void CompareCurve(ECCurve c1, ECCurve c2)
+        {
+            Assert.True((c1 == null && c2 == null) || (c1 != null && c2 != null));
+            if (c1 is ECNamedCurve)
+            {
+                ECNamedCurve curve1 = (ECNamedCurve)c1;
+                ECNamedCurve curve2 = (ECNamedCurve)c1;
+                Assert.Equal(curve1.Oid.FriendlyName, curve2.Oid.FriendlyName);
+            }
+            else if (c2 is ECPrimeCurve)
+            {
+                ECPrimeCurve curve1 = (ECPrimeCurve)c1;
+                ECPrimeCurve curve2 = (ECPrimeCurve)c2;
+                Assert.True(IsEqual(curve1.A, curve2.A));
+                Assert.True(IsEqual(curve1.B, curve2.B));
+                Assert.Equal(curve1.CurveType, curve2.CurveType);
+                Assert.True(IsEqual(curve1.G.X, curve2.G.X));
+                Assert.True(IsEqual(curve1.G.Y, curve2.G.Y));
+                Assert.True(IsEqual(curve1.H, curve2.H));
+                Assert.True(IsEqual(curve1.N, curve2.N));
+                Assert.True(IsEqual(curve1.P, curve2.P));
+                Assert.True(IsEqual(curve1.S, curve2.S));
+            }
+            else
+            {
+                Assert.True(false);
+            }
+        }
+
+        private static bool IsEqual(byte[] a, byte[] b)
```

Comment:
This is just Assert.Equal\<byte\>; which is generic-inference callable as Assert.Equal(a, b);